### PR TITLE
Xid_for_function_schedule_creation

### PR DESCRIPTION
**Pull request by CogniteJohan**
## Description
Please describe the change you have made.

## Checklist:
- [x] Tests added/updated.
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

**Review by CogniteJohan — COMMENTED** [review]
Added support to use the function external id when creating new schedules

File: cognite/client/_api/functions.py
Comment (haakonvt) [inline cognite/client/_api/functions.py:1]: Do I understand correctly that this is purely a client-side helper? (As in a request is first made to get the internal ID, then that is used when creating the schedule)

If yes, then I think we should be even more explicit in the docstring, and maybe even create a `Note:` section to make this distinction from the API crystal clear!

Comment (CogniteJohan) [reply cognite/client/_api/functions.py:1]: Added note, also added extra test if user tries to use both internal and external id.

File: tests/tests_unit/test_api/test_functions.py [hunk 1]
```diff
@@ -910,6 +907,17 @@ def test_list_schedules_with_function_id_and_function_external_id_raises(self, c
...
+    @pytest.mark.usefixtures("mock_function_schedules_response_with_xid")
+    def test_create_schedules_with_function_external_id(self, cognite_client):
+        res = cognite_client.functions.schedules.create(
+            name="my-schedule",
+            function_external_id="my-func",
+            cron_expression="*/5 * * * *",
+            description="Hi",
+        )
+
+        assert isinstance(res, FunctionSchedule)
```
Comment (haakonvt) [inline tests/tests_unit/test_api/test_functions.py:919]: I think you should also add a check here on the mock that `function_id` is what is passed to `_post` (and not external ID)

File: tests/tests_unit/test_api/test_functions.py
Comment (haakonvt) [inline tests/tests_unit/test_api/test_functions.py:1]: [suggestion] In file `tests/tests_integration/test_api/test_functions.py` you can also add an integration test that fails if the external ID does not exist

File: cognite/client/_api/functions.py
Comment (haakonvt) [reply cognite/client/_api/functions.py:1]: @CogniteJohan Where's this test?

**Review by HMEiding — COMMENTED** [review]
I'm a little bit on the fence about this change, as it might fool the user. So I agree, let's try to make it as clear as possible that the link is done by internal id under the hood.

File: cognite/client/_api/functions.py
[the remark this answers is not in the document]
Comment (CogniteJohan) [reply cognite/client/_api/functions.py:1]: Added

Comment (CogniteJohan) [reply cognite/client/_api/functions.py:1]: test_create_schedules_with_function_id_and_function_external_id_raises

File: tests/tests_unit/test_api/test_functions.py (hunk 1 above)
Comment (CogniteJohan) [reply tests/tests_unit/test_api/test_functions.py:933]: Done. Not sure if this is the right way to test variables passed but it seems to test the necessary functionality

File: tests/tests_unit/test_api/test_functions.py
Comment (CogniteJohan) [reply tests/tests_unit/test_api/test_functions.py:1]: Done

**Pull request merged**